blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,88 @@
+    url: https://github.com/sourcegraph/sourcegraph/issues/22912
+    category: Batch Changes
+  - description: "Code Insights drill-down filters now allow filtering insights data on the dashboard page using repo: filters."
+    url: https://github.com/sourcegraph/sourcegraph/issues/23186
+    category: Code Insights
+  - description: "Code Insights feature flag `DISABLE_CODE_INSIGHTS` environment variable has moved from the `repo-updater` service to the `worker` service. Any users of this flag will need to update their `worker` service configuration to continue using it."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23050
+    category: Repositories
+  - description: "The Docker Compose Jaeger container's `SAMPLING_STRATEGIES_FILE` now has a default value. If you are currently using a custom sampling strategies configuration, you may need to make sure your configuration is not overridden by the change when upgrading."
+    url: "https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/489"
+    category: Admin
+  - description: The copy icon displayed next to files and repositories will now copy the file or repository path. Previously, this action copied the URL to clipboard.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23390
+    category: Repositories
+  - description: "For Docker-Compose and Kubernetes users, the built-in main Postgres and codeintel databases have switched to an alpine Docker image. This requires re-indexing the entire database. This process can take up to a few hours on systems with large datasets."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23697
+    category: Repositories
+  - description: "All commit search results for `and`-expressions are now highlighted."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23336
+    category: Search
+  - description: Fixed a result streaming throttling issue that was causing significantly increased latency for some searches.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23736
+    category: Search
+  - description: Updating draft merge requests on GitLab from batch changes no longer removes the draft status.
+    url: https://github.com/sourcegraph/sourcegraph/issues/23944
+    category: Batch Changes
+  - description: Email notifications for saved searches are now deprecated in favor of Code Monitoring. Email notifications can no longer be enabled for saved searches. Saved searches that already have notifications enabled will continue to work, but there is now a button users can click to migrate to code monitors. Notifications for saved searches will be removed entirely in the future.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23275
+    category: Search
+  - description: "The experimental `codeInsightsAllRepos` feature flag has been removed."
```

Comment:
@iskyOS @Joelkw - I recommend cutting this update.

Reply:
Agreed